crates/era-cheatcodes/src/cheatcodes.rs

```diff
@@ -1533,13 +1621,21 @@ impl CheatcodeTracer {
             }
             getNonce_0(getNonce_0Call { account }) => {
                 tracing::info!("👷 Getting nonce for {account:?}");
-                let (account_nonce, _) =
-                    Self::get_nonce(account.to_h160(), &mut storage.borrow_mut());
-                tracing::info!(
-                    "👷 Nonces for account {:?} are {}",
-                    account,
-                    account_nonce.as_u64()
-                );
+                let account_nonce = if self.use_evm_fork.is_some() {
+                    let era_db = &storage.borrow_mut().storage_handle;
+                    let mut db = era_db.db.lock().unwrap();
+                    let info = db.basic(account).ok().flatten().expect("account info not found");
+                    U256::from(info.nonce)
```

Comment:
It's a bit weird, that we have to check use_evm_fork for all cheatcodes...
should we try to propogate it to their tracer?

Reply:
Actually we only need it for `warp`, `deal`, `getNonce`, `setNonce`, `etch` - and this is because these cheatcodes are directly calling the ZK storage. I believe this is the cleanest way to read the "evm" storage when in EVM mode.